functions/_common/explorerHandlers.ts

```diff
@@ -0,0 +1,216 @@
+        console.error(e)
+        return error(500, e)
+    }
+}
+
+export async function fetchCsvForExplorerView(
+    searchParams: URLSearchParams,
+    env: Env
+) {
+    const options = extractOptions(searchParams)
+
+    const url = env.url
+    url.href = url.href.replace(extensions.csv, "")
+
+    try {
+        const { grapherState } = await initGrapherForExplorerView(env, options)
+
+        console.log("checking if download is allowed")
+        ensureDownloadOfDataAllowed(grapherState)
+        console.log("data download is allowed")
+
+        const csv = assembleCsv(
+            grapherState,
+            searchParams ?? new URLSearchParams("")
+        )
+        return new Response(csv, {
+            headers: {
+                "Content-Type": "text/csv",
+            },
+        })
+    } catch (e) {
+        console.error(e)
+        return error(500, e)
+    }
+}
+
+export async function fetchMetadataForExplorerView(
+    searchParams: URLSearchParams,
+    env: Env
+) {
+    const options = extractOptions(searchParams)
+
+    const url = env.url
+    url.href = url.href.replace(extensions.metadata, "")
+
+    try {
+        const { grapherState } = await initGrapherForExplorerView(env, options)
+        const metadata = assembleMetadata(grapherState, searchParams)
+        return Response.json(metadata)
+    } catch (e) {
+        console.error(e)
+        return error(500, e)
+    }
+}
+
+export async function fetchReadmeForExplorerView(
+    searchParams: URLSearchParams,
+    env: Env
+) {
+    const options = extractOptions(searchParams)
+
+    const url = env.url
+    url.href = url.href.replace(extensions.readme, "")
+
+    try {
+        const { grapherState } = await initGrapherForExplorerView(env, options)
+        const readme = assembleReadme(grapherState, searchParams)
+        return new Response(readme, {
+            headers: {
+                "Content-Type": "text/markdown; charset=utf-8",
+            },
+        })
+    } catch (e) {
+        console.error(e)
+        return error(500, e)
+    }
+}
+
+export async function fetchZipForExplorerView(
+    searchParams: URLSearchParams,
+    env: Env
+) {
+    const options = extractOptions(searchParams)
+
+    const url = env.url
+    url.href = url.href.replace(extensions.zip, "")
+
+    try {
+        const { grapherState, explorerParams } =
+            await initGrapherForExplorerView(env, options)
+
+        ensureDownloadOfDataAllowed(grapherState)
+        const metadata = assembleMetadata(grapherState, searchParams)
+        const readme = assembleReadme(grapherState, searchParams)
+        const csv = assembleCsv(grapherState, searchParams)
+        console.log("Fetched the parts, creating zip file")
+
+        // Make a unique identifier for the given view
+        const explorerSlug = url.pathname.split("/").pop()
+        const viewId = Object.values(explorerParams)
+            .map((value) => value.replace(/\s/g, "_"))
+            .join("__")
```

Comment:
I tried to make it look similar to the mdim view ids, but it's a bit ugly.

Reply:
I wonder if @danyx23 didn't already introduce an explorer view ID in his explorer views work. I remember we talked about it, but not whether we really needed/wanted it. If so, we should use the same ID in both places, right?